### PR TITLE
binds: option to fullscreen pinned window

### PR DESCRIPTION
**Pull request by littleblack111**
#### Describe your PR, what does it fix/add?
option to fullscreen pinned window

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
nope

#### Is it ready for merging, or does it need work?
ready

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -534,6 +534,7 @@ CConfigManager::CConfigManager() {
     m_pConfig->addConfigValue("binds:movefocus_cycles_fullscreen", Hyprlang::INT{1});
     m_pConfig->addConfigValue("binds:disable_keybind_grabbing", Hyprlang::INT{0});
     m_pConfig->addConfigValue("binds:window_direction_monitor_fallback", Hyprlang::INT{1});
+    m_pConfig->addConfigValue("binds:pin_fullscreen", Hyprlang::INT{0});
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:537]: a name more like `allow_pin_fullscreen` makes more sense intuitively

+ missing documentation in configDescriptions.hpp

File: src/Compositor.cpp [hunk 2]
```diff
@@ -2352,7 +2353,21 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, sFullscreenS
     const eFullscreenMode CURRENT_EFFECTIVE_MODE = (eFullscreenMode)std::bit_floor((uint8_t)PWINDOW->m_sFullscreenState.internal);
     const eFullscreenMode EFFECTIVE_MODE         = (eFullscreenMode)std::bit_floor((uint8_t)state.internal);
 
-    const bool            CHANGEINTERNAL = !(PWINDOW->m_bPinned || CURRENT_EFFECTIVE_MODE == EFFECTIVE_MODE || (PWORKSPACE->m_bHasFullscreenWindow && !PWINDOW->isFullscreen()));
+	if (*PALLOWPINFULLSCREEN) {
```
Comment (vaxerski) [inline src/Compositor.cpp:2356]: can't these 2 ifs for *PALLOWPINFULLSCREEN be merged?

Comment (littleblack111) [reply src/Compositor.cpp:2356]: ah ya, i just try to not confuse my self when looking at the logic, fogot to put them back, thanks for reminding

**Review by vaxerski — APPROVED** [review]
lgtm, wiki mr needed

**Comment by littleblack111** [comment]
Please review the wiki mr

**Pull request merged**